### PR TITLE
Refactor qc for better understanding of error production. 

### PR DESCRIPTION
**Pull request by AEscobarCruz**
first refactor commit, nothing major just to push the branch up. created the string literals we shall use for errors instead of using rojoma-json internal library.

multiple changes to make the code a bit more clear and maintainable. fixed a few sytle errors but left many still open because not important right now. Error types are now in one location and so we can understand what errors are being passed where.

made one method into private.

changes made to QueryCoordinatorErrors and queryservice, which correspond to slight name changing.

slight clean up job.

### CODE REVIEW

**Comment by rozap** [comment]
lgtm :shipit:

**Comment by meganschneider** [comment]
Also looks good to me! :shipit:

File: query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+      schemaTimeout,
+      queryTimeout,
+      schemaCache,
+      schemaDecache,
+      secondaryInstance,
+      queryRewriter,
+      rollupInfoFetcher
+    )
+  }
+}
```
Comment (meganschneider) [inline query-coordinator/src/main/scala/com/socrata/querycoordinator/resources/QueryResource.scala]: ah, newline.

**Comment by AEscobarCruz** [comment]
fixed pr issue, merging now.

**Pull request merged**